bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Page.java

```diff
@@ -317,6 +318,30 @@ default boolean hasCloudconfigSupport() {
         throw new UnsupportedOperationException();
     }
 
+    /**
+     * URL to the root model of the App
+     *
+     * @return {@link String}
+     * @since com.adobe.cq.wcm.core.components.models 12.7.0
+     */
+    @Nullable
+    @JsonIgnore
+    default String getRootUrl() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Root page model of the current hierarchy of pages
+     *
+     * @return {@link Page}
+     * @since com.adobe.cq.wcm.core.components.models 12.7.0
+     */
+    @Nullable
+    @JsonIgnore
+    default Page getRootModel() {
```

Comment:
I'd rename it `getRootPageModel`

Reply:
as above, we propose `getHierarchyRootModel`